Shared/LocationTextController.cpp

```diff
@@ -128,25 +132,49 @@ void LocationTextController::onLocationChanged(const Point& pt)
 
   // update the elevation text
   if (m_useGpsForElevation)
+  {
     formatElevationText(pt.z());
-  else
+    return;
+  }
+
+  // abort if there's no valid elevation surface
+  if (!m_surface)
+    return;
+
+  // call the elevation function and pass the result to the format method
+  m_surface->elevationAsync(pt).then(this, [this](double elevation)
+  {
+    formatElevationText(elevation);
+  }).onFailed(this, [this](const ErrorException& e)
   {
-    // TODO:
-    // fixed a missing reference to the baseSurface possibly because the geoView changed signal
-    // was not emitted just by adding the .dted layer as an elevation source
-    // this should probably be updated to connect to the layers changed event
-    auto* scene = ToolResourceProvider::instance()->scene();
-    if (!scene)
-      return;
-    auto* surface = scene->baseSurface();
-    if (!surface)
-      return;
-
-    // set the elevation text in the continuation block
-    surface->elevationAsync(pt).then(this, [this](double elevation)
+    Q_UNUSED(e);
+    // multiple failures likely indicate a failed loading of the
+    // elevation source for the surface. set the member to nullptr
+    // to prevent continued calls for fetching the elevation
+    m_elevationFailures++;
+    if (m_elevationFailures > 2)
+      m_surface = nullptr;
```

Comment:
Is there any point in resetting `m_elevationFailures` here as well if we get into the condition? Assuming we keep this logic.